File: 0-unclassified/bigrace.cpp
#include <bits/stdc++.h>

using namespace std;

long long kpk(long long a, long long b) {
  return a / __gcd(a, b) * b;
}

int main() {
  long long t, w, b;
  cin >> t >> w >> b;
  long long a;
  if(kpk(b, w) < 0) {
    a = 0;
  } else {
    a = t / kpk(b, w);
  }
  cout << a << endl;
  long long x = kpk(b, w) * a;
  long long ans = a * min(b, w) + min(t - x + 1, min(b, w)) - 1;
  long long fpb = __gcd(ans, t);
  cout << ans << ' ' << t << endl;
  ans /= fpb;
  t /= fpb;
  printf("%I64d/%I64d\n", ans, t);
  return 0;
}

File: 0-unclassified/bigrace.java
import java.util.*;
import java.io.*;
import java.lang.*;
import java.math.BigInteger;

/**
 * Built using CHelper plug-in
 * Actual solution is at the top
 */

public class bigrace {
  public static void main(String[] args) {
    InputStream inputStream = System.in;
    OutputStream outputStream = System.out;
    InputReader in = new InputReader(inputStream);
    PrintWriter out = new PrintWriter(outputStream);
    Task solver = new Task();
    solver.solve(1, in, out);
    out.close();
  }
}

class Task { 
  long gcd(long a, long b) {
    return b != 0? gcd(b, a % b) : a;
  }
  
  long kpk(long a, long b) {
    return a / gcd(a, b) * b;
  }
  
  public void solve(int testNumber, InputReader in, PrintWriter out) {
    long t = in.nextLong();
    long w = in.nextLong();
    long b = in.nextLong();
    long a;
    BigInteger ret = BigInteger.valueOf(b / gcd(b, w));
    ret = ret.multiply(BigInteger.valueOf(w));
    if(ret.compareTo(BigInteger.valueOf(t)) == 1) {
      a = 0;
    } else {
      a = t / kpk(b, w);
    }
    long x = kpk(b, w) * a;
    long ans = a * Math.min(b, w) + Math.min(t - x + 1, Math.min(b, w)) - 1;
    long fpb = gcd(ans, t);
    ans /= fpb;
    t /= fpb;
    out.println(ans + "/" + t);
  }
}

class InputReader {
  public BufferedReader reader;
  public StringTokenizer tokenizer;

  public InputReader(InputStream stream) {
    reader = new BufferedReader(new InputStreamReader(stream), 32768);
    tokenizer = null;
  }

  public String next() {
    while (tokenizer == null || !tokenizer.hasMoreTokens()) {
      try {
        tokenizer = new StringTokenizer(reader.readLine());
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
    }
    return tokenizer.nextToken();
  }

  public int nextInt() {
    return Integer.parseInt(next());
  }

  public long nextLong() {
    return Long.parseLong(next());
  }
	
  public double nextDouble() {
    return Double.parseDouble(next());
  }
}